doc/source/train/overview.rst

```diff
@@ -0,0 +1,89 @@
+.. _train-key-concepts:
+
+.. _train-overview:
+
+Ray Train Overview
+==================
+
+.. image:: ./images/train-concepts.svg
+
+Ray Train has four key concepts:
+
+#. :ref:`Training function <train-overview-training-function>`: User-defined Python training loop.
+#. :ref:`Worker <train-overview-worker>`: Runs the training function.
+#. :ref:`Scaling configuration: <train-overview-scaling-config>` Configures the number of workers and the number of resources per worker.
+#. :ref:`Trainer <train-overview-trainers>`: Ties the training function, workers, and scaling configuration together to execute a single distributed training job.
+
+.. _train-overview-training-function:
+
+Training function
+-----------------
+
+The training function is a user-defined wrapper function that the Trainer dispatches to the distributed processes.
+It contains the training loop function with the addition of logic that provides the context that every worker needs to run the training job. 
+It loads the model, gets the shard of data, does checkpointing, and can have evaluation logic and metrics reporting.
+
+Ray Train documentation uses the following conventions:
+
+#. Users write a `train_func` function that defines training code.
+#. Users pass `train_func` to the Trainer through the `train_loop_per_worker` parameter because `train_func` executes the training loop on each worker process.
+
+.. code-block:: python
+
+    def train_func();
+    
+    # load model
+    ...
+    # load dataset
+    ...
+    # train model
+    ...
+    # report metrics and checkpoints
+    ...
```

Comment:
```suggestion
    def train_func():
        """User-defined training function that runs on each distributed worker process."""
        
        # Load model
        ...
        # Load dataset
        ...
        # Train model
        ...
        # Report metrics and checkpoints
        ...
```

Not the biggest fan of the comments/ellipses, but not sure how to make it cleaner and still convey the same point.

Reply:
Do you like any of these options better?
```
def train_func():
    """User-defined training function that runs on each distributed worker process."""

    # Load model
    model = load_model()
    
    # Load dataset
    dataset = load_dataset()
    
    # Train model
    train_model(model, dataset)
    
    # Report metrics and checkpoints
    report_metrics_and_checkpoints(model)
```

OR

```
def train_func():
    """User-defined training function that runs on each distributed worker process."""

    # Load model
    TODO: Load model code here
    
    # Load dataset
    TODO: Load dataset code here
    
    # Train model
    CODE: Train model code here
    
    # Report metrics and checkpoints
    CODE: Report metrics and checkpoints code here
```